share/spack/setup-env.sh

```diff
@@ -84,7 +84,10 @@ function spack {
             if [ "$_sp_arg" = "-h" ]; then
                 command spack cd -h
             else
-                cd $(spack location $_sp_arg "$@")
+		LOC="$(spack location $_sp_arg "$@")"
+		if [[ "x" -ne "x$LOC" ]] ; then
```

Comment:
Agreed on "-d", I'll put in an updated patch for that.


________________________________
From: Adam J. Stewart
Sent: Friday, April 29, 2016 7:26:37 AM
To: LLNL/spack
Cc: Scogland, Tom; Author
Subject: Re: [LLNL/spack] check that a target directory exists before committing to cd (#861)


In share/spack/setup-env.sh<https://github.com/LLNL/spack/pull/861#discussion_r61585624>:

> @@ -84,7 +84,10 @@ function spack {
>              if [ "$_sp_arg" = "-h" ]; then
>                  command spack cd -h
>              else
> -                cd $(spack location $_sp_arg "$@")
> +             LOC="$(spack location $_sp_arg "$@")"
> +             if [[ "x" -ne "x$LOC" ]] ; then


if [[ $LOC ]]; then

or even

if [[ -d $LOC ]]; then

would be preferable. See man test

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/LLNL/spack/pull/861/files/10d8ed4d98314f298c5fd28d726834129510fbd5#r61585624>